geode-core/src/integrationTest/java/org/apache/geode/management/DataQueryEngineIntegrationTest.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.cache.query.data.Portfolio;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.management.internal.ManagementConstants;
+import org.apache.geode.management.internal.SystemManagementService;
+import org.apache.geode.management.internal.beans.DataQueryEngine;
+import org.apache.geode.management.internal.cli.json.TypedJson;
+import org.apache.geode.management.model.EmptyObject;
+import org.apache.geode.management.model.Item;
+import org.apache.geode.management.model.Order;
+import org.apache.geode.management.model.SubOrder;
+import org.apache.geode.pdx.PdxInstance;
+import org.apache.geode.pdx.PdxInstanceFactory;
+import org.apache.geode.pdx.internal.PdxInstanceFactoryImpl;
+import org.apache.geode.test.junit.categories.GfshTest;
+import org.apache.geode.test.junit.rules.ServerStarterRule;
+
+/**
+ * Functional integration tests for {@link DataQueryEngine}.
+ */
+@Category({GfshTest.class})
+public class DataQueryEngineIntegrationTest {
```

Comment:
I think this class might end up being brittle in the various tests that are checking large JSON strings since there probably is no guarantee of order in the JSON? I would prefer to see these JSON strings extracted as resources and then also use the `json-path` library to validate the content (or at least parts of it). Although it's coupled to `spring-test` you can see some use of `json-path` in `RestAccessControllerTest`. Lastly, please switch to using Jackson instead of `org.json`.

Reply:
yes, that's why some tests in this suite still lack assert statement. I just want to add some initial assertions first before I start ripping out the org.json code.

Reply:
Sounds good.